### PR TITLE
User-Onboarding: Added initial CreateAccount Modal

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6641

## Description of Changes
- Adds initial step of CreateAccount Modal

## "How We Fixed It"
N/A

## Test Plan
This PR only adds new components, so not much to test.

- Enable `FLAG_USER_ONBOARDING_ENABLED` in `.env`.
- Add `type='create-account'` to `<AuthModal/>` instance in `SubLayoutHeader.tsx` and open the modal, verify it matches [figma](https://www.figma.com/file/I5jbkBPzNrbYMHYQJb6CNY/Onboarding?type=design&node-id=802-5224&mode=design&t=zakjlow5RcFRlS0c-4) or apply this diff
```diff
diff --git a/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx b/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
index 4564d26340..c58647b700 100644
--- a/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
+++ b/packages/commonwealth/client/scripts/views/components/SublayoutHeader/SublayoutHeader.tsx
@@ -59,6 +59,7 @@ export const SublayoutHeader = ({
         open={isFeedbackModalOpen}
       />
       <AuthModal
+        type='create-account'
         onClose={() => setIsAuthModalOpen(false)}
         isOpen={isAuthModalOpen}
       />

```

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

**Review by burtonator — DISMISSED** [review]

File: packages/commonwealth/client/scripts/views/modals/AuthModal/types.ts [hunk 1]
```diff
@@ -6,7 +6,11 @@ export type AuthModalTabs = {
   options: AuthWallets[] | AuthSSOs[];
 };
 
-export type BaseModalProps = {
+export type CreateAccountModal = {
+  onClose: () => any;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/AuthModal/types.ts:10]: let's do not use any

File: packages/commonwealth/client/scripts/views/modals/AuthModal/common/ModalLayout/ModalLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    showExistingAccountSignInFooter: false,
+  },
+};
+
+export type ModalLayoutProps = {
+  onClose: () => any;
+  type: 'create-account' | 'sign-in';
+  body: ReactNode;
+  bodyClassName?: string;
+  onSignInClick?: () => any;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/modals/AuthModal/common/ModalLayout/ModalLayout.tsx:30]: let's not use any, please

**Review by masvelio — APPROVED** [review]
1. add padding for `Common is built on web3 technology that utilizes wallets.
How would you like to sign up?` text on mobile

image

2. please do not use `any` type

File: packages/commonwealth/client/scripts/views/modals/AuthModal/common/ModalLayout/ModalLayout.tsx (hunk 2 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/AuthModal/common/ModalLayout/ModalLayout.tsx:30]: updated.

File: packages/commonwealth/client/scripts/views/modals/AuthModal/types.ts (hunk 1 above)
Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/modals/AuthModal/types.ts]: updated.

**Comment by mzparacha** [comment]
Changes applied, please re-review.

cc: @masvelio @burtonator

**Review by burtonator — APPROVED** [review]

**Review by burtonator — APPROVED** [review]

**Review by masvelio — APPROVED** [review]

**Comment by mzparacha** [comment]
image


seems it can't be merged without master branch re-sync?

cc: @burtonator

**Pull request merged**